bus_board_part2.rb

```diff
@@ -0,0 +1,100 @@
+require 'net/http'
+require 'uri'
+require 'json'
+
+module WebRequest
+  def web_request(uri)
+
+    response = Net::HTTP.get_response(uri)
+
+    JSON.parse(response.body)
+
+  end
+end
+
+class BusStop
+  include WebRequest
+
+  attr_accessor :stop_id, :buses_info
+
+  def initialize(stop_id)
+    @stop_id = stop_id
+    @buses_info = nil
+  end
+
+  def fetch_buses_info
+    uri = URI("https://api.tfl.gov.uk/StopPoint/#{@stop_id}/Arrivals")
+    @buses_info = web_request(uri)
+  end
+
+  def bus_prediction(number_of_buses)
+    sorted_buses = @buses_info.sort_by { |bus| bus["timeToStation"] }

```

Comment:
At the moment whenever we use `@buses_info` we sort it first. Admittedly we only use it once so it doesn't make a difference, but if in the future we add more features they'll probably want the buses sorted by when they arrive too. Is there anything we can do to reduce how often we need to sort this list?